stm/stm.ml

```diff
@@ -2503,12 +2542,12 @@ let process_transaction ?(newtip=Stateid.fresh ())
             (VCS.branches ());
           VCS.checkout_shallowest_proof_branch ();
           VCS.commit id (Alias (oid,x));
-          Backtrack.record (); if w == VtNow then finish (); `Ok
+          Backtrack.record (); if w == VtNow then ignore(finish ~doc:0); `Ok
```

Comment:
the `0` is wrong, process transaction is clearly a function that takes a doc and gives a doc.

I see this patch is not "final" and only the external APIs are functional.
Ithink you should define a `let dummy_doc = 0` up there in the file, use it instead of 0, and then, later, remove the definition of `dummy_doc` and let the compiler complain.

Reply:
Very good point thanks; will do. The story is that I have a patch really making `VCS` fucntional, but it is huge and should be splitted. The point here is that this "dummy" approach already helped me to track and solve some coqtop bugs; the hope is that at least now we can make clients be more explicit wrt what they are doing.

Reply:
The difficulty of the patch removing `VCS` is indeed that the worker API needs some modifications; other than that it makes for a very interesting adventure.